test/attempts.test.js

```diff
@@ -23,4 +24,24 @@ describe('Request attempts', function () {
       done();
     });
   });
+
+  it('should call delay strategy 2 times after some retries', function (done) {
+    var mockDelayStrategy = sinon.stub().returns(100);
+    var startTime = process.hrtime();
+    request({
+      url: 'http://www.filltext.com/?rows=1&err=500', // return a 500 status
+      maxAttempts: 3,
+      retryDelay: 1000000, // Set to large delay to prove it will not be used
+      delayStrategy: mockDelayStrategy
+    }, function (err, response, body) {
+      var endTime = process.hrtime(startTime); // process.hrtime returns an array [seconds, nanoseconds]
+      var timeDiff = endTime[0] * 1000 + endTime[1]/1000000;
+
+      // If mockDelayStrategy returns 1ms, timeDiff would be ~560ms. Since it returns 100ms, we expect the timeDiff to be between 700 and 800 inclusive.
+      t.isTrue(timeDiff >= 700 && timeDiff <= 800, 'delayStrategy\'s return value was not used or was used incorrectly');
```

Comment:
Good call. I changed the test method a bit to use isAtLeast. My computer takes ~560ms just to run the code without delay. Therefore, I adjusted the delay time returned to be 500ms so that if the delay code ran, the run time should be at least 1000ms. With the previous test method, I was likely to cause circleci to fail because it would run much faster.